src/coreclr/tools/Common/Compiler/DependencyAnalysis/Relocation.cs

```diff
@@ -333,6 +335,47 @@ private static unsafe int GetLoongArch64PC12(uint* pCode)
             return imm;
         }
 
+        private static unsafe int GetRiscv64PC12(uint* pCode)
+        {
+            uint auipcInstr = *pCode;
+
+            // first get the high 20 bits,
+            int imm = (int)(((auipcInstr >> 12) & 0xFFFFF) << 12);
```

Comment:
I think it is okay like `auipcInstr & 0xFFFFF000` (no shifts)

Reply:
Done